src/vehicle.cpp

```diff
@@ -6488,31 +6466,12 @@ int vehicle::manual_fire_turret( int p, player &shooter, item &gun )
         // Put our shooter on the roof of the vehicle
         shooter.add_effect( effect_on_roof, 1 );
 
-        int to_fire = abs(parts[p].mode) > 1 ? std::max( gun.gun_get_mode( "AUTO" ).qty, 1 ) : 1;
-        res = shooter.fire_gun( targ, to_fire, gun );
+        int to_fire = abs(parts[p].mode) > 1 ? std::max( gun->gun_get_mode( "AUTO" ).qty, 1 ) : 1;
```

Comment:
There should be a better way of getting the biggest burst mode out there.

`gun_get_mode( "AUTO" )` would fail in case of burst fire (non-auto) guns and default to 1. This is obviously wrong.
Even worse - it prevents NPCs from using burst fire weapons that aren't full auto.